### ISSUE TITLE
Code review

### ISSUE THREAD

**Issue by GreatTwang** [duqinaerfa/EC601#2]
Hi! I reviewed your code many days ago and I am sorry to put my code review here just now.

In general, your code works well. You successfully implemented all the features required in this mini project with only one file. It impressed me that you completed this task step by step and reflect your progress on the github. Besides, your code is quite readable with the help of sufficient comments.

On the other hand, I have several suggestions for you. First, your Readme file can be more concrete. Installation guidance is important so that others can reproduce your results quickly.  Some instructions for usage should also be included to help others understand the project better. Second, you can make your code more modular and design better APIs so that others can use your library more conveniently.

In conclusion, you did a good job for mini project 1 and it is my pleasure to do your code review!
